AX: HTML menu element should map to role=list

annevk opened the pull request:
#### af6778aa0405885a7638a1ef60e195cf050483a9
<pre>
AX: HTML menu element should map to role=list
<a href="https://bugs.webkit.org/show_bug.cgi?id=201581">https://bugs.webkit.org/show_bug.cgi?id=201581</a>
rdar://55145117

Reviewed by Tyler Wilcock.

As far as I can tell we only have a special code path remaining for
&quot;type=touchbar&quot;, not &quot;toolbar&quot;, thus removing that.

Otherwise this generally makes the menu element behave like an ul
element for accessibility purposes.

This builds on work in #3571 by SirriCelles.

* LayoutTests/accessibility/display-contents-element-roles-expected.txt:
* LayoutTests/accessibility/mac/menu-type-toolbar-expected.txt: Removed.
* LayoutTests/accessibility/mac/menu-type-toolbar.html: Removed.
* LayoutTests/accessibility/roles-exposed.html:
* LayoutTests/platform/glib/accessibility/aria-visible-element-roles-expected.txt:
* LayoutTests/platform/glib/accessibility/display-contents-element-roles-expected.txt:
* LayoutTests/platform/ios/accessibility/display-contents-element-roles-expected.txt:
* LayoutTests/platform/mac-wk1/accessibility/aria-visible-element-roles-expected.txt:
* LayoutTests/platform/mac-wk1/accessibility/roles-exposed-expected.txt:
* LayoutTests/platform/mac-wk2/accessibility/aria-visible-element-roles-expected.txt:
* LayoutTests/platform/mac-wk2/accessibility/roles-exposed-expected.txt:
* LayoutTests/platform/wpe/accessibility/aria-visible-element-roles-expected.txt:
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::AXObjectCache::createObjectFromRenderer):
* Source/WebCore/accessibility/AccessibilityList.cpp:
(WebCore::AccessibilityList::isUnorderedList const):
* Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
(WebCore::AccessibilityNodeObject::determineAccessibilityRoleFromNode const):
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
(WebCore::AccessibilityRenderObject::inheritsPresentationalRole const):

Canonical link: <a href="https://commits.webkit.org/264077@main">https://commits.webkit.org/264077@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/944c2247c9da6e784aac04ea76bc154cccfc501a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/6618 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/6834 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/7015 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/8208 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/6903 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/6617 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/7891 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/6782 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/9795 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/6735 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/7250 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/6054 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/8294 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/4721 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/5986 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/13827 "2 flakes 1 missing results 155 failures") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/6538 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/6066 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/8735 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/6555 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/5369 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/5952 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/5914 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1566 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/10122 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/6324 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on LayoutTests/accessibility/roles-exposed.html:
```diff
@@ -6,7 +6,7 @@
 
 <!-- ==================================================================================================== -->
-<!-- HTML elements in alphabetical order, excepting the need for nesting (e.g. <li> is with its <ol> and <ul> parents) -->
+<!-- HTML elements in alphabetical order, excepting the need for nesting (e.g., <li> is with its <meu>, <ol>, and <ul> parents) -->
```
```suggestion
<!-- HTML elements in alphabetical order, excepting the need for nesting (e.g., <li> is with its <menu>, <ol>, and <ul> parents) -->
```

twilco reviewed (approved):
Oops, forgot about that, sorry! LGTM.

annevk committed:
AX: HTML menu element should map to role=list
https://bugs.webkit.org/show_bug.cgi?id=201581
rdar://55145117

Reviewed by Tyler Wilcock.

As far as I can tell we only have a special code path remaining for
"type=touchbar", not "toolbar", thus removing that.

Otherwise this generally makes the menu element behave like an ul
element for accessibility purposes.

This builds on work in #3571 by SirriCelles.

* LayoutTests/accessibility/display-contents-element-roles-expected.txt:
* LayoutTests/accessibility/mac/menu-type-toolbar-expected.txt: Removed.
* LayoutTests/accessibility/mac/menu-type-toolbar.html: Removed.
* LayoutTests/accessibility/roles-exposed.html:
* LayoutTests/platform/glib/accessibility/aria-visible-element-roles-expected.txt:
* LayoutTests/platform/glib/accessibility/display-contents-element-roles-expected.txt:
* LayoutTests/platform/ios/accessibility/display-contents-element-roles-expected.txt:
* LayoutTests/platform/mac-wk1/accessibility/aria-visible-element-roles-expected.txt:
* LayoutTests/platform/mac-wk1/accessibility/roles-exposed-expected.txt:
* LayoutTests/platform/mac-wk2/accessibility/aria-visible-element-roles-expected.txt:
* LayoutTests/platform/mac-wk2/accessibility/roles-exposed-expected.txt:
* LayoutTests/platform/wpe/accessibility/aria-visible-element-roles-expected.txt:
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::AXObjectCache::createObjectFromRenderer):
* Source/WebCore/accessibility/AccessibilityList.cpp:
(WebCore::AccessibilityList::isUnorderedList const):
* Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
(WebCore::AccessibilityNodeObject::determineAccessibilityRoleFromNode const):
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
(WebCore::AccessibilityRenderObject::inheritsPresentationalRole const):

Canonical link: https://commits.webkit.org/264077@main
